[WPE][GTK] Add webkitgtk-test-fonts to WebKit source tree

obyknovenius opened the pull request:
#### 3cb36fcf23dd31fc488d7c59298e5306ae201d74
<pre>
[WPE][GTK] Add webkitgtk-test-fonts to WebKit source tree
<a href="https://bugs.webkit.org/show_bug.cgi?id=256252">https://bugs.webkit.org/show_bug.cgi?id=256252</a>

Reviewed by Adrian Perez de Castro.

This ensures that the fonts required for testing on GTK and WPE will
always be available at a predictable location, regardless of the
building environment.

* Tools/WebKitTestRunner/InjectedBundle/gtk/ActivateFontsGtk.cpp:
(WTR::getFontsPath):
(WTR::initializeFontConfigSetting):
(WTR::getOutputDir): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/wpe/ActivateFontsWPE.cpp:
(WTR::getFontsPath):
(WTR::activateFonts):
(WTR::topLevelPath): Deleted.
(WTR::getOutputDir): Deleted.
* Tools/buildstream/elements/test-infra.bst:
* Tools/buildstream/elements/test-infra/webkitgtk-test-fonts.bst: Removed.
* Tools/gtk/jhbuild.modules:
* Tools/wpe/jhbuild.modules:

Canonical link: <a href="https://commits.webkit.org/263863@main">https://commits.webkit.org/263863@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/efccd7b9adc4502ba851e99026dcc1da93c8a9c9

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/5933 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/6108 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/6294 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/7486 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/6292 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/5930 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/6327 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/6067 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/8713 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/6041 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/6078 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/5375 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/7546 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/6037 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/3555 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/5351 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/13283 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/5420 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/5430 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/7647 "run-api-tests-without-change (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/5879 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/4823 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/5311 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1404 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/9437 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/5672 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

clopez commented on Tools/WebKitTestRunner/InjectedBundle/gtk/ActivateFontsGtk.cpp:
```diff
@@ -34,37 +34,19 @@
-        return webkitOutputDir;
-
-    CString topLevelPath = WTR::topLevelPath();
-    GUniquePtr<char> outputDir(g_build_filename(topLevelPath.data(), "WebKitBuild", nullptr));
```
BTW, if we ship those files on the WebKit repo maybe we don't even need to install them. Perhaps we can point the tooling to pick them from `g_build_filename(topLevelPath.data(), "Tools", "glib", "fonts," nullptr)`

Just an idea.

obyknovenius replied:
We already have `webkitFontsConfDir` that points to this directory. No additional path evaluation is needed.

aperezdc reviewed (commented):
I like the idea of having the fonts in the WebKit tree. The path should be `Tools/WebKitTestRunner/glib/fonts/`, though: the WPE port also uses them, they are not GTK specific. Other than that, the patch LGTM.

aperezdc reviewed (approved):
Nice, thanks for the fixups @obyknovenius!

obyknovenius committed:
[WPE][GTK] Add webkitgtk-test-fonts to WebKit source tree
https://bugs.webkit.org/show_bug.cgi?id=256252

Reviewed by Adrian Perez de Castro.

This ensures that the fonts required for testing on GTK and WPE will
always be available at a predictable location, regardless of the
building environment.

* Tools/WebKitTestRunner/InjectedBundle/gtk/ActivateFontsGtk.cpp:
(WTR::getFontsPath):
(WTR::initializeFontConfigSetting):
(WTR::getOutputDir): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/wpe/ActivateFontsWPE.cpp:
(WTR::getFontsPath):
(WTR::activateFonts):
(WTR::topLevelPath): Deleted.
(WTR::getOutputDir): Deleted.
* Tools/buildstream/elements/test-infra.bst:
* Tools/buildstream/elements/test-infra/webkitgtk-test-fonts.bst: Removed.
* Tools/gtk/jhbuild.modules:
* Tools/wpe/jhbuild.modules:

Canonical link: https://commits.webkit.org/263863@main
